WordPress/Classes/ViewRelated/Blog/SharingAuthorizationWebViewController.swift

```diff
@@ -145,43 +156,51 @@ class SharingAuthorizationWebViewController: WPWebViewController {
     // MARK: - URL Interpretation
 
     private func authorizeAction(from url: URL) -> AuthorizeAction {
-        let requested = url.absoluteString
-
         // Path oauth declines are handled by a redirect to a path.com URL, so check this first.
-        if requested.range(of: SharingAuthorizationWebViewController.declinePath) != nil {
+        if AuthorizeURLComponents.declinePath.containedIn(url) {
             return .deny
         }
 
-        if !requested.hasPrefix(SharingAuthorizationWebViewController.authorizationPrefix) {
+        if !url.absoluteString.hasPrefix(AuthorizeURLComponents.authorizationPrefix.rawValue) {
             return .none
         }
 
-        if requested.range(of: SharingAuthorizationWebViewController.requestActionParameter) != nil {
+        if AuthorizeURLComponents.requestActionParameter.containedIn(url) {
             return .request
         }
 
         // Check the rest of the various decline ranges
-        if requested.range(of: SharingAuthorizationWebViewController.denyActionParameter) != nil {
+        if AuthorizeURLComponents.denyActionParameter.containedIn(url) {
             return .deny
         }
 
         // LinkedIn
-        if requested.range(of: SharingAuthorizationWebViewController.userRefused) != nil {
+        if AuthorizeURLComponents.userRefused.containedIn(url) {
             return .deny
         }
 
         // Facebook and Google+
-        if requested.range(of: SharingAuthorizationWebViewController.accessDenied) != nil {
+        if AuthorizeURLComponents.accessDenied.containedIn(url) {
             return .deny
         }
 
         // If we've made it this far and verifyRange is found then we're *probably*
         // verifying the oauth request.  There are edge cases ( :cough: tumblr :cough: )
         // where verification is declined and we get a false positive.
-        if requested.range(of: SharingAuthorizationWebViewController.verifyActionParameter) != nil {
+        if AuthorizeURLComponents.verifyActionParameter.containedIn(url) {
+            return .verify
+        }
+
+        // Facebook
+        if AuthorizeURLComponents.state.containedIn(url) && AuthorizeURLComponents.code.containedIn(url) {
```

Comment:
This is not a blocker, just a possible improvement.

The URL format I see is:
```
https://public-api.wordpress.com/connect/?code=VERY_LONG_RANDOM_STRING&state=VERY_LONG_RANDOM_STRING#_=_
```
I'm not sure if failure cases can also include these random strings (tokens). If they can, it seems possible (though somewhat unlikely) for theses tokens to sometimes contain the characters "code" and "state", causing a false positive here.

Perhaps looking for query params with the names "code" and "state" would work more reliably:
```
        if let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: true)?.queryItems {
            let containsState = queryItems.contains(where: { queryItem in
                queryItem.name == AuthorizeURLComponents.state.rawValue
            })
            let containsCode = queryItems.contains(where: { queryItem in
                queryItem.name == AuthorizeURLComponents.code.rawValue
            })
            if containsState && containsCode {
                return .verify
            }
        }
```